community/io/src/main/java/org/neo4j/io/pagecache/impl/muninn/MuninnPageCursor.java

```diff
@@ -158,6 +172,8 @@ void clearPageState()
         pointer = victimPage; // make all future page access go to the victim page
         pageSize = 0; // make all future bound checks fail
         page = null; // make all future page navigation fail
+        currentPageId = UNBOUND_PAGE_ID;
+        cursorException = null;
```

Comment:
wdyt about just checking cursorException is really null and throw exception otherwise instead? to completely avoid cases of lost exception